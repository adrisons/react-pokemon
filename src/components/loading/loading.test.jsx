import React from "react";
import renderer from "react-test-renderer";
import pokeball from "../../assets/pokeball.png";
import Loading from "./loading";
import { render, screen } from "@testing-library/react";

describe("WHEN: Loading", () => {
  const component = renderer.create(<Loading />);
  const tree = component.toJSON();
  it("THEN: should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });

  it("THEN: img must have src = pokeball.png and alt = loading", () => {
    render(<Loading />);
    const logo = screen.getByRole("img");
    expect(logo).toHaveAttribute("src", pokeball);
    expect(logo).toHaveAttribute("alt", "loading");
  });
});
