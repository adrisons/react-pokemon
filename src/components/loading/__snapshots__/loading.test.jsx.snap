// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WHEN: Loading THEN: should match snapshot 1`] = `
<div
  className="loading"
>
  <img
    alt="loading"
    src="pokeball.png"
  />
</div>
`;
